### ISSUE TITLE
Issue with overlapping matches

### ISSUE THREAD

**Issue by sondr3** [maciejhirsz/logos#55]
As mentioned in #40, I'm playing around with Logos myself right now and just hit this, I have 

```rust
    #[regex = "cm|mm|Q|in|pc|pt|px"]
    AbsoluteLength,
    #[regex = "[-a-zA-Z_][a-zA-Z0-9_-]*"]
    Ident,
```

as some of my tokens in a toy CSS parser I'm writing, and all the absolute lengths works fine... except for `Q`, which gets interpreted as `Ident` instead of `AbsoluteLength` in this test.

```rust
        assert_lex(
            "h3 { letter-spacing: 1Q } /* quarter-millimeters */",
            &[
                (Token::Ident, "h3"),
                (Token::CurlyBracketOpen, "{"),
                (Token::Ident, "letter-spacing"),
                (Token::Colon, ":"),
                (Token::Number, "1"),
                (Token::AbsoluteLength, "Q"),
                (Token::CurlyBracketClose, "}"),
            ],
        );
```

Curiously, all the other absolute lengths work fine, just not the single letter `Q`. Any idea why? You can also try it out for yourself at https://github.com/sondr3/minifie-rs.

**Comment by maciejhirsz** [comment]
This should be fixed now with 0.9.5 just released on crates.io.

**Comment by sondr3** [comment]
Yep, that fixed it. Sweet, thanks!
